experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -684,9 +667,9 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents>, AnchorLoca
 					start: 0,
 					end: count,
 				});
-				this.withParentNodeUpToRoot((p) => {
-					p.events.emit("beforeChange", p);
-				});
+				if (this.parent instanceof PathNode) {
```

Comment:
Nit: would it be worth adding a local helper function that predicates the event emission based on whether or not the parent is a PathNode? That would also give a convenient place to explain why we only emit in that case.